### ISSUE TITLE
Not respecting required nullable properly on trivial objects

### ISSUE THREAD

**Issue by filipbekic01** [astahmer/openapi-zod-client#203]
In next schema, `parent_id` is not set as nullable. It's required but it should be nullable.

```
{
  "openapi": "3.0.0",
  "info": {
    "title": "Etsy Seller Taxonomy API",
    "version": "1.0.0"
  },
  "paths": {},
  "components": {
    "schemas": {
      "TestClass": {
        "type": "object",
        "properties": {
          "id": {
            "type": "integer",
            "format": "int32"
          },
          "parent_id": {
            "type": "integer",
            "format": "int32",
            "nullable": true
          },
          "children": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/TestClass"
            }
          }
        },
        "required": [
          "id",
          "parent_id"
        ]
      }
    }
  }
}
```

**Comment by astahmer** [comment]
the zod output seems fine here, the TS type might need `| null`
https://openapi-zod-client.vercel.app/?doc=N4KABBYEQPYA4FMB2BDOBLKAuaBmAdAAxFQA04kUcKALgBYDO2Yokb0A9IjUzq%2BwKgBzBDWb8Bk6ACcEDODCQM54ilPXQAJggBmKAK4AbMXzUbzUAMaKayEyzPmn0NHEPpLtdIo4ArBoqqzsHsUAyWdAgAtihBIfFsUAAksjrMUADEHNZRCkh2DBzhkTGFACpyNADChigMTI4JGgC%2BjU1sre3sne09wX3qAx1mzeSJOXkFcYLF0XXT6lAVDNW19QtOUDQAnojpMABGvgiWYmMhVNLwCNI06CqmXVDomhvtW7sI6ehItiLSZDaCSgOhg0hi9mev1wACYoECnKMERZqLJfgB9F5vJofPY4KF-G6ArqJUHg2jfaFw84k6BIIy1A6GL44GjSfQIZEtGnAiLoQyaNHY4E7PEuaTSFDbYm06DoWxRXgOWWhFK6dJZCaKApFCJzcqVGp1BoqoYhM3crmhWQAR306FkrxwAG0rVJnq8efEqCg0TRMa83ZAALpWi1gAatZpAA&prettier=EQbwOgdgBDVsAHATgSwgFwOooCboBbwBcUAjAEwAMANJLHMOgIYBG2ehwJALLdLPCZIkAewDuABSEBTCAGdiDJgBsxTAJ4LgfevACuc6QBVWWkgDMVhnQOCGAtikXoke6TZjw5aAObLpAIp6IujSipbK1nS2Lkwoyr4AwiL29kyK8NJyAKzwkAC%2BwEA

feel free to send a PR

**Comment by astahmer** [comment]
just fixed it here if you want
https://typed-openapi-astahmer.vercel.app/

**Comment by filipbekic01** [comment]
I actually made PR here https://github.com/astahmer/openapi-zod-client/pull/204

We need null properties, undefined is something else.

**Comment by astahmer** [comment]
> We need null properties, undefined is something else.

hm, yes ? typed-openapi outputs this (which is the exactly the same as you expected in your OP)
`export type TestClass = { id: number; parent_id: number | null; children?: Array<TestClass> | undefined };`

**Comment by filipbekic01** [comment]
> > We need null properties, undefined is something else.
> 
> hm, yes ? typed-openapi outputs this (which is the exactly the same as you expected in your OP) `export type TestClass = { id: number; parent_id: number | null; children?: Array<TestClass> | undefined };`

You're right, I checked wrong commit before. I tried the same as you did but has some TypeScript errors but I didnt want to use @ts-ignore there. Your getTs() method has different return type I assume.

Anyways, this nullable field seems to be deprecated in OpenAPI 3.1 version?

**Comment by astahmer** [comment]
> Anyways, this nullable field seems to be deprecated in OpenAPI 3.1 version?

yes it seems to be the case, I also just noticed when implementing it in typed-openapi
